File: algorithms/swift/1-two_sum.swift
// https://leetcode.com/problems/two-sum/

class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var map = [Int: Int]()
        for i in 0..<nums.count {
            if let found = map[target - nums[i]] {
                return [found, i]
            }
            map[nums[i]] = i
        }
        return [Int]()
    }
}

File: algorithms/c++/1-two-sum.cpp
// https://leetcode.com/problems/two-sum/
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> mp;
        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];
            if (mp.find(complement) != mp.end()) {
                return vector<int> {mp[complement], i};
            } 
            mp[nums[i]] = i;
        }
        return vector<int>{};
    }
};

int main() {

    return 0;
}